feat: playground v1

k11kirky opened the pull request:
V0.1 for LLM o11y playground

k11kirky committed:
feat: playground v1

github-actions[bot] committed:
Update UI snapshots for `chromium` (6)

github-actions[bot] committed:
Update UI snapshots for `chromium` (6)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

k11kirky committed:
merge

k11kirky committed:
feat: playground v1

k11kirky committed:
fix: extract models ready to use proxy

k11kirky committed:
Merge branch 'feat-llmo/playground' of github.com:PostHog/posthog into feat-llmo/playground

k11kirky committed:
fix: greptile issues

k11kirky committed:
fix: playground

k11kirky committed:
Merge branch 'master' into feat-llmo/playground

k11kirky committed:
Merge branch 'master' into feat-llmo/playground

k11kirky committed:
feat: use proxy

k11kirky committed:
Merge branch 'master' into feat-llmo/playground

k11kirky committed:
feat: version 2 of playground

k11kirky committed:
feat: ui update

k11kirky committed:
add comment

k11kirky committed:
feat: cleanup pr 2

k11kirky committed:
Merge branch 'master' into feat-llmo/playground

github-actions[bot] committed:
Update UI snapshots for `chromium` (8)

k11kirky committed:
fix when playground shows

k11kirky committed:
Merge branch 'feat-llmo/playground' of github.com:PostHog/posthog into feat-llmo/playground

github-actions[bot] committed:
Update UI snapshots for `chromium` (8)

kappa90 reviewed (changes requested):
<img width="149" alt="image" src="https://github.com/user-attachments/assets/dc5271d6-ff47-485d-b339-f6d3341c442a" />

I found this Config button confusing, it's basically a toggle to show/hide parts of the UI, but is shown as a primary button, which might be misleading.

Also, if you try to Run the prompt but you haven't added any message, nothing happens. The button should probably be disabled unless you already have messages in the Input, or you should be prompted to add some. The messages also need to be non-empty, otherwise it won't run.

If you put 0 max tokens, you still get an answer, for a very simple reason: I didn't implement max tokens support in the proxy 😅  so that number doesn't do anything, it's not sent to the API.

Great otherwise, I left a couple comments in the code that I'd fix before releasing this (mainly the API key authentication one).

k11kirky committed:
Merge branch 'master' into feat-llmo/playground

k11kirky committed:
feat: add max_tokens and temp

k11kirky committed:
feat: add ff and address comments

k11kirky commented:
Hey @kappa90 addressed your comments and throw this behind a feature flag :) let me know if any other questions. There is a whole log of UI changes (including config button) in a future PR so that was ignored here for now. Best bet is to roll it out for us and get feedback

github-actions[bot] committed:
Update UI snapshots for `chromium` (6)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (5)

github-actions[bot] committed:
Update UI snapshots for `chromium` (5)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

kappa90 commented on products/editor/backend/api/proxy.py:
```diff
@@ -33,12 +38,16 @@
     AnthropicConfig.SUPPORTED_MODELS_WITH_THINKING + OpenAIConfig.SUPPORTED_MODELS_WITH_THINKING
 )
 
+ALLOWED_FEATURE_FLAGS = ["llm-editor-proxy", "llm-observability-playground"]
```
Minor: this is not used anymore

kappa90 reviewed (approved):
All great, just some UX comments:
- Still confused by the "Toggle Config" button, it doesn't seem to add a lot to the UX.
- Would also put the "Add Message" button in a more prominent position, possibly on top (I had the toolbar on and it was covered, plus I had to scroll to find it, plus I had to scroll A LOT to find it once I got a response from the agent in the AI Output panel).
- Would have expected the "AI Output" bit to disappear when added to the Chat History, if the AI Output is long you end up with two walls of text one next to another.
- Would be nice to have a button within the chat messages, to add them to the "Compare" section (in case I forgot to do it before)

k11kirky committed:
feat: address comments

k11kirky committed:
Merge branch 'feat-llmo/playground' of github.com:PostHog/posthog into feat-llmo/playground

k11kirky committed:
Merge branch 'master' into feat-llmo/playground

k11kirky committed:
Merge remote-tracking branch 'origin' into feat-llmo/playground

k11kirky committed:
costs

k11kirky committed:
Merge branch 'feat-llmo/playground' of github.com:PostHog/posthog into feat-llmo/playground

k11kirky committed:
snapshot

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

k11kirky committed:
fix: tests

k11kirky committed:
fix: tests

k11kirky committed:
Merge branch 'master' into feat-llmo/playground

k11kirky committed:
Merge branch 'master' into feat-llmo/playground
